libs/MeshKernel/tests/src/RangeCheckTests.cpp

```diff
@@ -0,0 +1,170 @@
+#include "MeshKernel/Exceptions.hpp"
+#include "MeshKernel/RangeCheck.hpp"
+
+#include <limits>
+#include <string>
+#include <utility>
+
+#include <gtest/gtest.h>
+
+using namespace meshkernel;
+
+template <typename T>
+class RangeCheckFixture : public testing::Test
+{
+public:
+    void CheckEqual()
+    {
+        EXPECT_NO_THROW(range_check::CheckEqual<T>(1, 1, name));
+        EXPECT_THROW(range_check::CheckEqual<T>(1, 2, name), RangeError);
+
+        if (std::same_as<T, float> ||
+            std::same_as<T, double> ||
+            std::same_as<T, long double>)
+        {
+            double constexpr x = 1.0;
+            double constexpr epsilon = std::numeric_limits<T>::epsilon();
```

Comment:
Replace double with T, otherwise only testing for double

Very minor (even less than very minor), since long double is there what about std::float128_t (not sure if this is c++23 only)(equivalent to __float128)

Reply:
Yes, saw thiose, in fact this->CheckEqual() and this->CheckNotEqual() were never called in the typed test.

[std::is_floating_point](https://en.cppreference.com/w/cpp/types/is_floating_point) , which is used in the concept std::floating_point, covers std::float128_t (sice c++23). I did not check the outcome of the test though.